RulerDetectionPlugin/src/RulerSettingsDialog.cpp

```diff
@@ -58,145 +59,126 @@ namespace nmc {
+		horizontalLayout_3->addWidget(refPoints_label);
 
-		comboBox_tickDistance = new QComboBox(centralWidget);
-		comboBox_tickDistance->setObjectName(QStringLiteral("comboBox_tickDistance"));
-
-		horizontalLayout_3->addWidget(comboBox_tickDistance);
+		refPoints_cb = new QComboBox(this);
+		refPoints_cb->setObjectName(QStringLiteral("refPoints_cb"));
 
+		horizontalLayout_3->addWidget(refPoints_cb);
 		verticalLayout->addLayout(horizontalLayout_3);
 
+		// Set hessian threshold
 		horizontalLayout_4 = new QHBoxLayout();
 		horizontalLayout_4->setSpacing(6);
 		horizontalLayout_4->setObjectName(QStringLiteral("horizontalLayout_4"));
-		label_3 = new QLabel(centralWidget);
-		label_3->setObjectName(QStringLiteral("label_3"));
-
-		horizontalLayout_4->addWidget(label_3);
-
-		comboBox_referencePoints = new QComboBox(centralWidget);
-		comboBox_referencePoints->setObjectName(QStringLiteral("comboBox_referencePoints"));
-
-		horizontalLayout_4->addWidget(comboBox_referencePoints);
-
+		hessianThres_label = new QLabel(this);
+		hessianThres_label->setObjectName(QStringLiteral("hessianThres_label"));
+		horizontalLayout_4->addWidget(hessianThres_label);
+
+		hessianThres_sb = new QSpinBox(this);
+		hessianThres_sb->setObjectName(QStringLiteral("hessianThres_sb"));
+		hessianThres_sb->setMinimum(10);
+		hessianThres_sb->setMaximum(1500);
+		hessianThres_sb->setSingleStep(10);
+		hessianThres_sb->setValue(200);
+
+		horizontalLayout_4->addWidget(hessianThres_sb);
 		verticalLayout->addLayout(horizontalLayout_4);
 
+		// ### Add buttons and connect SIGNALs to SLOTS
 		horizontalLayout_5 = new QHBoxLayout();
 		horizontalLayout_5->setSpacing(6);
 		horizontalLayout_5->setObjectName(QStringLiteral("horizontalLayout_5"));
-		label_4 = new QLabel(centralWidget);
-		label_4->setObjectName(QStringLiteral("label_4"));
+		help_btn = new QPushButton(this);
+		help_btn->setObjectName(QStringLiteral("help_btn"));
+		connect(help_btn, SIGNAL(clicked()), this, SLOT(helpPressed()));
 
-		horizontalLayout_5->addWidget(label_4);
+		horizontalLayout_5->addWidget(help_btn);
 
-		spinBox_HessianThreshold = new QSpinBox(centralWidget);
-		spinBox_HessianThreshold->setObjectName(QStringLiteral("spinBox_HessianThreshold"));
-		spinBox_HessianThreshold->setMinimum(10);
-		spinBox_HessianThreshold->setMaximum(1000);
-		spinBox_HessianThreshold->setSingleStep(10);
-		spinBox_HessianThreshold->setValue(150);
+		OK_btn = new QPushButton(this);
+		OK_btn->setObjectName(QStringLiteral("OK_btn"));
+		connect(OK_btn, SIGNAL(clicked()), this, SLOT(okPressed()));
+		horizontalLayout_5->addWidget(OK_btn);
 
-		horizontalLayout_5->addWidget(spinBox_HessianThreshold);
+		cancel_btn = new QPushButton(this);
+		cancel_btn->setObjectName(QStringLiteral("cancel_btn"));
+		connect(cancel_btn, SIGNAL(clicked()), this, SLOT(cancelPressed()));
+		horizontalLayout_5->addWidget(cancel_btn);
 
 		verticalLayout->addLayout(horizontalLayout_5);
```

Comment:
- `QGridLayout` is still be the best choice for this dialog - you can combine it with a `QVBoxLayout` for the buttons. Moreover, you can use RowSpan/ColumnSpan when adding a widget. 
- `QDialogButtonBox` is the standard way to create dialog buttons. advantage: OK/Cancel are positioned correctly for different OS (plus you don't need that many lines : )
- `setAlignment` fixes the issues when resizing the dialog (setAlignment(Qt::AlignLeft | Qt::AlignTop) would probably be the best choice here - the labels should be right aligned